### ISSUE TITLE
New Rule: `no-invalid-hex-colors`

### ISSUE THREAD

**Issue by lumirlumir** [eslint/css#244]
### Rule details

Disallow invalid hex colors.

### What type of rule is this?

Warns about a potential problem

### Example code

```css
/* Not OK */

a { color: #00; }

a { color: #fff1az; }

a { color: #12345aa; }

/* OK */

a { color: #000; }

a { color: #000f; }

a { color: #fff1a0; }

a { color: #123450aa; }
```

### Prior Art

```markdown
stylelint: https://stylelint.io/user-guide/rules/color-no-invalid-hex
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Longhand hex colors can be either 6 or 8 (with alpha channel) hexadecimal characters. And their shorthand variants are 3 and 4 characters respectively.

So, if a hex color does not follow these formats, it won't be recognized as a valid hex color.

**Comment by lumirlumir** [comment]
I've looked into it and found that it works well with named properties. 

However, it doesn't work for custom properties as shown below:

```css
/* eslint css/no-invalid-properties: "error" */

:root {
	--color-1: #1;
	--color-2: #12;
	--color-3: #12345;
	--color-4: #1234567;
}
```

The existing Stylelint rule checks for custom properties:

<img width="696" height="148" alt="Image" src="https://github.com/user-attachments/assets/2ddc9622-e05b-47ed-9c87-75b9f89fd431" />

I wonder if another CSS rule can check this behavior?

**Comment by nzakas** [comment]
Custom properties are always parsed as `Raw` because we can't know what type they are until they're used. As @xbinaryx points out, once you try to use the custom property in a property value, you'll get a validation error.
